src/doc/style-guide/src/expressions.md

```diff
@@ -536,6 +540,70 @@ if a_long_expression
 {
     ...
 }
+
+if let Some(a) = b
+    && another_long_expression
+    || a_third_long_expression
+{
+    ...
+}
+
+if let Some(relatively_long_thing)
+    = a_long_expression
+    && another_long_expression
+    || a_third_long_expression
+{
```

Comment:
I think this is a good case for breaking _after_ the assignment operator (rustfmt's current, technically buggy behavior in these control flow expressions, as well as the current behavior and rules for statements). I feel this is technically consistent with current rules, but IMO it would benefit from having the different indentation/spacing for the `a_long_expression` line instead of the assignment operator in the `let` expr being the same indent as the binops for the subsequent clauses in the chain